flexmeasures/data/models/planning/tests/test_solver.py

```diff
@@ -272,25 +272,49 @@ def test_fallback_to_unsolvable_problem(target_soc, charging_station_name):
     )
 
 
+@pytest.mark.parametrize(
+    "market_scenario",
+    [
+        "dynamic contract",
+        "fixed contract",
+    ],
+)
 def test_building_solver_day_2(
     db,
     add_battery_assets,
     add_market_prices,
+    create_test_tariffs,
     add_inflexible_device_forecasts,
     inflexible_devices,
     flexible_devices,
+    market_scenario: str,
 ):
-    """Check battery scheduling results within the context of a building with PV, for day 2,
-    which is set up with 8 expensive, then 8 cheap, then again 8 expensive hours.
-    We expect the scheduler to:
-    - completely discharge within the first 8 hours
-    - completely charge within the next 8 hours
-    - completely discharge within the last 8 hours
+    """Check battery scheduling results within the context of a building with PV, for day 2, against the following market scenarios:
+    1) a dynamic tariff with equal consumption and feed-in tariffs, that is set up with 8 expensive, then 8 cheap, then again 8 expensive hours.
+    2) a fixed consumption tariff and a fixed feed-in tariff that is lower.
+    In the test data:
+    - Hours with net production coincide with low dynamic market prices.
+    - Hours with net consumption coincide with high dynamic market prices.
+    In either scenario, we expect the scheduler to:
+    - completely discharge within the first 8 hours (either due to 1) high prices, or 2) net consumption)
+    - completely charge within the next 8 hours (either due to 1) low prices, or 2) net production)
+    - completely discharge within the last 8 hours (either due to 1) high prices, or 2) net consumption)
     """
-    epex_da = Sensor.query.filter(Sensor.name == "epex_da").one_or_none()
     battery = flexible_devices["battery power sensor"]
     building = battery.generic_asset
-    assert battery.get_attribute("market_id") == epex_da.id
+    if market_scenario == "dynamic contract":
+        consumption_price_sensor = Sensor.query.filter(
+            Sensor.name == "epex_da"
+        ).one_or_none()
+        production_price_sensor = consumption_price_sensor
+        assert battery.get_attribute("market_id") == consumption_price_sensor.id
```

Comment:
Isn't this assumption useful in both scenarios?

Reply:
Fair point.